### Import Pandas and Numpy libraries

In [26]:
import pandas as pd
import numpy as np

from matplotlib import pyplot as plt
import seaborn as sns

### Import csv files for initial analysis

In [29]:
observations = pd.read_csv('observations.csv', encoding='utf-8')
observations.head()

scientific_name                            park_name  observations
0        Vicia benghalensis  Great Smoky Mountains National Park            68
1            Neovison vison  Great Smoky Mountains National Park            77
2         Prunus subcordata               Yosemite National Park           138
3      Abutilon theophrasti                  Bryce National Park            84
4  Githopsis specularioides  Great Smoky Mountains National Park            85

In [30]:
species = pd.read_csv('species_info.csv')
species.head()

category                scientific_name  \
0   Mammal  Clethrionomys gapperi gapperi   
1   Mammal                      Bos bison   
2   Mammal                     Bos taurus   
3   Mammal                     Ovis aries   
4   Mammal                 Cervus elaphus   

                                        common_names conservation_status  
0                           Gapper's Red-Backed Vole                 NaN  
1                              American Bison, Bison                 NaN  
2  Aurochs, Aurochs, Domestic Cattle (Feral), Dom...                 NaN  
3  Domestic Sheep, Mouflon, Red Sheep, Sheep (Feral)                 NaN  
4                                      Wapiti Or Elk                 NaN

### Check columns/variables of each file and number of rows

In [32]:
print(observations.columns)
print(species.columns)

Index(['scientific_name', 'park_name', 'observations'], dtype='object')
Index(['category', 'scientific_name', 'common_names', 'conservation_status'], dtype='object')


### Length of data contained in each file:

In [33]:
print('The length of observations.csv is: ' + str(len(observations)))
print('The length of species.csv is: ' + str(len(species)))

The length of observations.csv is: 23296
The length of species.csv is: 5824


### Data type contained in each column:

In [34]:
print(observations.dtypes)
print(species.dtypes)

scientific_name    object
park_name          object
observations        int64
dtype: object
category               object
scientific_name        object
common_names           object
conservation_status    object
dtype: object


## Initial Analysis of Data:

### Two CSV files: 
### Observations.csv containing 23296 rows of data, with 3 columns: 
- scientific_name - (object, str, with nominal data)
- park_name - (object, str, with nominal data)
- observations - (integer, Quantitative, discrete counts).

### Species.csv containing 5824 rows of data, with 4 columns: 
- category (object, str, nominal)
- scientific_name (object, str, nominal)
- common_names (object, str, nominal)
- conservation_status (object, str, ordinal)

# Project Goals:

### In this project the perspective will be through a biodiversity analyst for the National Parks Service. The National Park Service wants to ensure the survival of at-risk species, to maintain the level of biodiversity within their parks. Therefore, the main objectives as an analyst will be understanding characteristics about the species and their conservations status, and those species and their relationship to the national parks. Some questions that are posed:

- What is the distribution of conservation status for species?
- Are certain types of species more likely to be endangered?
- Are the differences between species and their conservation status significant?
- Which animal is most prevalent and what is their distribution amongst parks?